logger.js

```diff
@@ -85,58 +85,56 @@ function pinoLogger (opts, stream) {
   delete opts.useLevel
 
   const genReqId = reqIdGenFactory(opts.genReqId)
-  loggingMiddleware.logger = logger
-  return loggingMiddleware
-
-  function onResFinished (err) {
-    this.removeListener('close', onResFinished)
-    this.removeListener('error', onResFinished)
-    this.removeListener('finish', onResFinished)
+  const result = (req, res, next) => {
```

Comment:
instead of attaching the pino logger to the statically defined loggingMiddleware function and returning the single reference to it, we create a new closure called result and attach the pino logger to the result. Each call to pinoHttp will create a new result closure so each call gets it's own closure rather than sharing the loggingMiddleware function and it's logging property.